algorithms/scripts/RIT1-PCA+LDA-htcai.py

```diff
@@ -0,0 +1,219 @@
+
+# coding: utf-8
+
+# # Create a logistic regression model to predict RIT1 mutation from gene expression data in TCGA
+
+# In[1]:
+
+import os
+import urllib
+import random
+import warnings
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn import preprocessing
+from sklearn.linear_model import SGDClassifier
+from sklearn.model_selection import train_test_split, GridSearchCV
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.preprocessing import StandardScaler
+from sklearn.decomposition import PCA
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
+
+
+# In[2]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# ## Specify model configuration
+
+# In[3]:
+
+# We're going to be building a 'RIT1' classifier 
+GENE = '6016'# RIT1
+
+
+# *Here is some [documentation](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) regarding the classifier and hyperparameters*
+# 
+# *Here is some [information](https://ghr.nlm.nih.gov/gene/RIT1) about RIT1*
+
+# ## Load Data
+
+# In[4]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('..', 'download', 'expression-matrix.tsv.bz2')\nX = pd.read_table(path, index_col=0)")
+
+
+# In[5]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('..', 'download', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
+
+
+# In[6]:
+
+y = Y[GENE]
+
+
+# In[7]:
+
+# The Series now holds RIT1 Mutation Status for each Sample
+y.head(6)
+
+
+# In[8]:
+
+# Here are the percentage of tumors with RIT1
+y.value_counts(True)
+
+
+# ## Set aside 10% of the data for testing
+
+# In[9]:
+
+# Typically, this can only be done where the number of mutations is large enough
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
```

Comment:
I decided to change the size of the testing data to 0.2 and set `stratify=y`, which probably is able to make the testing score more reliable.